src/rust/src/packed_r_tree.rs

```diff
@@ -637,45 +606,69 @@ impl PackedRTree {
                             range: HttpRange::RangeFrom(start..),
                         });
                     }
-                    continue;
-                }
-
-                let combine_request_node_threshold =
-                    combine_request_threshold / size_of::<NodeItem>();
-
-                // Add node to search recursion
-                match (queue.back_mut(), node_item.offset as usize) {
```

Comment:
I broke this complex match statement into a series of dumber if blocks. It's not a little more redundant, but I think easier to follow.